server/init.lua

```diff
@@ -23,17 +26,53 @@ event.listen("component_added", updateMachineList)
 
 local statuses = {multiblocks = {}, power = {}}
 
+local function updateMachineStatuses(_evName, _localAddress, sender, port, _distance, ...)
+    local args = {...}
+    if port == portNumber and args[1] == "local_multiblock_statuses" then
+        for address, status in pairs(serialization.unserialize(args[2])) do
+            statuses.multiblocks[address] = status
+        end
+    end
+end
+event.listen("modem_message", updateMachineStatuses)
+
+local function updatePowerStatus(_evName, _localAddress, sender, port, _distance, ...)
+    local args = {...}
+    if port == portNumber and args[1] == "local_power_status" then
+        statuses.powerStatus = serialization.unserialize(args[2])
+    end
+end
+event.listen("modem_message", updatePowerStatus)
+
 function server.configure(x, y)
     -- TODO: Code for GUI configuration of machines:
     ---- Machine renaming
     ---- Machine widgets layout
 end
 
+function server.load()
+    -- TODO: Ping other servers for their statuses on boot up
+    ---- Check if there's no response and determine if this is the main server
+end
+
 function server.update()
+    local shouldBroadcastStatuses = false
     for address, name in pairs(machineAddresses) do
-        statuses.multiblocks[address] = getMultiblockStatus(address, name)
+        local multiblockStatus = getMultiblockStatus(address, name)
+        shouldBroadcastStatuses = statuses.multiblocks[address].state ~= multiblockStatus.state
+        statuses.multiblocks[address] = multiblockStatus
+    end
+    if shouldBroadcastStatuses then
+        modem.broadcast(portNumber, "local_multiblock_statuses", serialization.serialize(statuses.multiblocks))
     end
-    statuses.power = getPowerStatus(powerAddress, "Lapotronic Supercapacitor")
+
+    if powerAddress then
+        local powerStatus = getPowerStatus(powerAddress, "Lapotronic Supercapacitor")
+        if statuses.powerStatus ~= powerStatus then
+            modem.broadcast(portNumber, "local_power_status", serialization.serialize(powerStatus))
```

Comment:
This code will almost never run: Local servers either 1. Have no access to a battery storage and 2. Don't need to show the status on the HUD, making sending the data to the main server fairly useless.

Reply:
I was thinking
What if the battery storage is not on the main server and is in a local server for someone?

Reply:
Then we provide no support.
It's not feasible to send data through a network at the rate the HUD should update.

Power monitor is pretty much the only thing on the HUD that actually _needs_ to be on the main server.